### ISSUE TITLE
Use node name  (wsgi) instead of trace_id as basis for sorting

### ISSUE THREAD

**Issue by shuwens** [lkatalin/Categorization#3]

**Comment by shwsun** [comment]
@shwsun please take a look at this

**Comment by lkatalin** [comment]
Update: I have fixed the output of `new_json_dag.py` to use this syntax. Output example:

```
$ python new_json_dag.py samples/d702d7bb-c717-4c28-95b1-b05078ef0f65.json

 # 1 R: 3678 usecs 
Digraph {
	wsgi_osapi_volume    [label="wsgi - osapi_volume"]
	rpc_cinder-scheduler [label="rpc - cinder-scheduler"]
	rpc_cinder-volume    [label="rpc - cinder-volume"]
	driver_cinder-volume [label="driver - cinder-volume"]
	wsgi_osapi_volume    [label="wsgi - osapi_volume"]
	wsgi_osapi_volume    [label="wsgi - osapi_volume"]
	wsgi_osapi_volume -> rpc_cinder-scheduler [label="-1 day, 23:59:59.986313"]
	rpc_cinder-scheduler -> rpc_cinder-volume [label="0:00:00.009620"]
	rpc_cinder-volume -> driver_cinder-volume [label="-1 day, 23:59:58.978254"]
	wsgi_osapi_volume -> wsgi_osapi_volume    [label="0:00:02.040393"]
	wsgi_osapi_volume -> wsgi_osapi_volume    [label="0:00:00.017613"]
}
```

The node labels are now the same as the node names, which is redundant, but I am leaving it for now. Next I will look at the `main.py` code to make sure it can process this (put it into groups).

**Comment by lkatalin** [comment]
In `make_dag.py`, the `id_to_obj` hash table and the name of each node in the DAG are now both based on `id_str`, ex: `rpc_cinder-volume`. Previously, `id_to_obj` keys were trace_ids and nodes were created based on the .dot label, ex `rpc - cinder-volume`. Basically, this makes things more consistent.

These changes are necessary because of the way `main.py` creates a traversable DAG from the .dot file. Previously this DAG was based on the trace_id.

**Comment by lkatalin** [comment]
![infinte_loop](https://user-images.githubusercontent.com/22075926/31791075-81fbc64c-b4e5-11e7-8ab0-6780dbf78e4e.png)
TL;DR the new design is susceptible to infinite loops due to the re-use of node names within a trace. Brainstorming how to prevent this.

I think we want to do this on the .dot level (in `new_json_dag.py` conversion) so that we can visualize the DAGs. But I believe the only way to do that would be to change node names, ex. `wsgi_osapi_volume` becomes `wsgi_osapi_volume_01`. @shwsun @RS1999ent any thoughts?

**Comment by lkatalin** [comment]
Closing this issue because we had decided to fix this problem by basing our .dot model on the trace_ids, but using the node names (labels) for grouping. See #10.
